replicatedhq/ship
# Make ship work without git installed, resolving #484

What I Did
------------
added back the github client removed in pr #479 to resolve #484
also refactored fetch code to make adding new sources a little less cluttered in the future

How I Did it
------------
I reintroduced the github client removed in https://github.com/replicatedhq/ship/commit/5509af69d8186a08f167fd8f35e129d2db987ffd and have it set to be used preferentially (unless the url is a go-getter url such as `github.com/replicatedhq/test-charts?ref=3427d6997bd150c60caa00ba0298fdfe17e3ed04//plain-k8s`)

How to verify it
------------
Run `ship init` on a system without git. To confirm go-getter still works, you can run with a go-getter compatible ref: `ship init github.com/replicatedhq/test-charts?ref=3427d6997bd150c60caa00ba0298fdfe17e3ed04//plain-k8s` or look at the newly added integration test `plain-k8s-gogetter`.

Description for the Changelog
------------
`ship init` no longer requires git to be installed to access github repos.


Picture of a Boat (not required but encouraged)
------------

![USS John Finn (DDG-113)](https://upload.wikimedia.org/wikipedia/commons/4/4a/USS_John_Finn_%28DDG-113%29_arrives_at_Pearl_Harbor_on_10_July_2017.JPG "USS John Finn (DDG-113)")
